sklearn/linear_model/_logistic.py

```diff
@@ -873,6 +932,11 @@ def _logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
                     'Initialization coef is of shape %d, expected shape '
                     '%d or %d' % (coef.size, n_features, w0.size))
             w0[:coef.size] = coef
+            if solver == 'lbfgs' and precondition:
+                if fit_intercept:
+                    w0[-1] += np.inner(w0[:n_features], X_mean)
```

Comment:
why inner here and dot later?

Reply:
the result of this here is a number. I think ``np.inner`` makes this more clear.